daemon/mgr/container_validation.go

```diff
@@ -91,6 +93,46 @@ func (mgr *ContainerManager) validateConfig(c *Container, update bool) ([]string
 	return warnings, nil
 }
 
+// validateDiskQuota is used to validate disk quota config
+func (mgr *ContainerManager) validateDiskQuota(config *types.ContainerCreateConfig) error {
+	if config == nil {
+		return errors.Errorf("invalid request, create config is nil")
+	}
+
+	if config.DiskQuota == nil {
+		if config.QuotaID != "" && config.QuotaID != "0" {
+			return errors.Wrap(errInvalidDiskQuota, "set QuotaID without DiskQuota")
+		}
+		return nil
+	}
+
+	quota := config.DiskQuota
+	if len(quota) > 1 && config.QuotaID != "" {
+		return errors.Wrap(errInvalidDiskQuota, `QuotaID only used to set one disk quota, `+
+			`such as: "/=10G" or "/path1=10G" or ".*=10G"`)
+	}
+
+	for key := range quota {
+		if key == "" {
+			return errors.Wrap(errInvalidDiskQuota, "quota can not be nil string")
+		}
+
+		paths := strings.Split(key, "&")
+		if len(paths) <= 1 {
```

Comment:
should we check for one path?

Reply:
one path maybe regular expression, so no good idea to check it.

Reply:
got it.